protocol/datasync/transport.go

```diff
@@ -165,6 +166,8 @@ func splitPayloadInBatches(payload *protobuf.Payload, maxSizeBytes int) []*proto
 
 // CalculateSendTime calculates the next epoch
 // at which a message should be sent.
+// We randomize it a bit so that not all messages are sent on the same epoch
 func CalculateSendTime(count uint64, time int64) int64 {
-	return time + int64(uint64(math.Exp2(float64(count-1)))*backoffInterval*offsetToSecond)
+	return time + int64(uint64(math.Exp2(float64(count-1)))*backoffInterval*offsetToSecond) + int64(rand.Intn(30)) // nolint: gosec
+
```

Comment:
Just a little point about `math/rand`, if you want the result to be random you need to set a source of randomness, otherwise your results will always be the same.

See https://gobyexample.com/random-numbers